plugin/src/main/kotlin/com/vanniktech/maven/publish/SonatypeHost.kt

```diff
@@ -28,4 +26,15 @@ enum class SonatypeHost(
       "$rootUrl/service/local/staging/deployByRepositoryId/${stagingRepositoryId.get()}/"
     }
   }
+
+  companion object {
+    fun valueOf(sonatypeHost: String): SonatypeHost = when (sonatypeHost) {
+      "DEFAULT" -> DEFAULT
+      "S01" -> S01
+      else -> throw IllegalArgumentException("No SonatypeHost constant $sonatypeHost")
+    }
+
+    val DEFAULT = SonatypeHost("https://oss.sonatype.org")
+    val S01 = SonatypeHost("https://s01.oss.sonatype.org")
```

Comment:
Could you add `@JvmStatic` and `@JvmField` to the function and properties respectively? That should keep compatibility for Groovy users.